tests/core/tools_utils/test_oauth_tool_connector_eviction.py

```
@@ -0,0 +1,89 @@
+        "https://auth.example/authorize", user_id="u1"
+    )
+
+
+def test_401_clears_stale_user_tools(patched_manager):
+    """Fix #2: after 401, _user_tools must be cleared so the _connect placeholder
+    is re-exposed and the LLM stops calling dead tools."""
+    patched_manager._store = MagicMock(spec=DalTokenStore)
+
+    connector = OAuthToolConnector()
+    toolset = _make_toolset()
+    stale_tool = MagicMock(); stale_tool.name = "k8s_list_pods"; stale_tool.toolset = toolset
```

Ruff (0.15.17)

[error] Multiple statements on one line (semicolon)

(E702)